### DISCUSSION TITLE
Ensoniq ASR10 "superdisk" in .img format, trying to get working

### DISCUSSION THREAD

**Discussion by danwarren1971** [keirf/flashfloppy#947] [General]
Hi There, On my Ensoniq ASR10, there appear two formats of superdisk - .hfe and .img files. I'm a linux user where .hfe support is sparse, and using the epslin project to create the disk images and add the libraries to them.
Standard images and HFE superdisks seem to work well, but the .img files are proving problematic - without any custom cfg I get "error 31" when selecting them 

I've then tried to work out the values to customise IMG.CFG, so using fdisk utility and the source code of epslin(neo), I've established that sectors are 512, sectors = 10200  and total filesize is 5222400, In the epslin sources, I can see also that ASR uses 20 sectors per track, but my attached IMG.CFG causes the drive to hang up - so I'm stuck! 

I'm not really a software guy, and I'm not even sure if what I'm trying to achieve is possible - posting a sample image and my configs in the hope that I might get some pointers!

[ASR-10_SUPERDISK.zip](https://github.com/user-attachments/files/17327139/ASR-10_SUPERDISK.zip)

**Comment by keirf** [comment]
Share a SuperDisk HFE image, as the format can be worked out from that.

**Reply by danwarren1971** [reply]
Thanks Keir, 

Just been double checking, and I may have got ahead of myself with what I said in the first post - turns out the HFE format isn't working well either - I saw repeatable 'file read errors' and 'data corrupted' type messages on the Ensoniq while trying to save bigger files, but nothing reporting an issue on the gotek itself. I was able to save one file to the disk image attached, which was a very small soundbank definition file, PCM files for samples are larger and repeatably fail to save.

Standard 1.6m disk size images are working rock stable, I've attached one of those also if it's any help

[ASRblank-hfe.zip](https://github.com/user-attachments/files/17328410/ASRblank-hfe.zip)
[STANDARD_IMG.zip](https://github.com/user-attachments/files/17328484/STANDARD_IMG.zip)

**Reply by keirf** [reply]
That's a regular 1.6MB HFE image, not a SuperDisk HFE image. I need the latter, please.

Goteks with 415 chip are unreliable when writing to high-density HFE image files. You really need to use IMG files.

**Reply by keirf** [reply]
Actually I suspect that a SuperDisk is a normal high-density disk but with 255 cylinders instead of 80.

**Reply by keirf** [reply]
Please unzip and try the attached IMG.CFG file with your SuperDisk IMG files.

[IMG.CFG.zip](https://github.com/user-attachments/files/17334000/IMG.CFG.zip)

**Reply by danwarren1971** [reply]
Thanks Keir, I think I see my mistake was having sectors and tracks values transposed, this seems to work now! I've just loaded a couple of 5.1Mb images and they loaded in just fine  -  Really happy to see it working finally!

I'll do some more testing in the morning, but looking very good. Thank you so much with your support today.. above and beyond!

**Reply by danwarren1971** [reply]
Confirming it all works perfectly with that IMG.CFG, thanks Keir

**Reply by keirf** [reply]
Great. This is now merged ready for next  release.
